src/libsyntax/parse/parser.rs

```diff
@@ -4698,7 +4698,7 @@ impl<'a> Parser<'a> {
                         let (fields, etc) = self.parse_pat_fields().unwrap_or_else(|mut e| {
                             e.emit();
                             self.recover_stmt();
-                            (vec![], false)
+                            (vec![], true)
```

Comment:
Testing my understanding: this is getting assigned to the name `etc`; so, the mechanism here is that we're pretending that struct patterns that failed to parse have a `..` so that we shouldn't emit additional errors about missing fields?

Reply:
Correct.